quickwit-search/src/search_stream/leaf.rs

```diff
@@ -207,6 +236,139 @@ fn collect_fast_field_values(
     Ok(buffer)
 }
 
+async fn leaf_search_stream_single_split_partitionned<'a>(
+    split: SplitIdAndFooterOffsets,
+    index_config: Arc<dyn IndexConfig>,
+    stream_request: &SearchStreamRequest,
+    index: Index,
+    split_schema: Schema,
+    fast_field: TypedFastField<'a>,
+    partition_fast_field: TypedFastField<'a>,
+) -> crate::Result<LeafSearchStreamResult> {
+    let fast_field_collector_builder = PartionnedFastFieldCollectorBuilder::new(
+        fast_field.field_type.value_type(),
+        fast_field.field_name.to_string(),
+        index_config.timestamp_field_name(),
+        index_config.timestamp_field(&split_schema),
+        stream_request.start_timestamp,
+        stream_request.end_timestamp,
+        partition_fast_field.field_type.value_type(),
+        partition_fast_field.field_name.to_string(),
+    )?;
+
+    let output_format = OutputFormat::from_i32(stream_request.output_format).ok_or_else(|| {
+        SearchError::InternalError("Invalid output format specified.".to_string())
+    })?;
+
+    if output_format != OutputFormat::PartitionnedClickhouseRowBinary {
+        return Err(SearchError::InternalError(
+            "Invalid output format specified, only partitionnedRowBinary is allowed when you provide a parition-by field.".to_string(),
+        ));
+    }
+
+    let search_request = SearchRequest::from(stream_request.clone());
+    let query = index_config.query(split_schema, &search_request)?;
+
+    let reader = index
+        .reader_builder()
+        .num_searchers(1)
+        .reload_policy(ReloadPolicy::Manual)
+        .try_into()?;
+
+    let searcher = reader.searcher();
+    warmup(
+        &*searcher,
+        query.as_ref(),
+        &fast_field_collector_builder.fast_field_to_warm(),
+    )
+    .await?;
+    let mut buffer = Vec::new();
+    match fast_field_collector_builder.value_type() {
+        (Type::I64, Type::I64) => {
+            let fast_field_collector = fast_field_collector_builder.build_i64_i64();
+            let fast_field_values = searcher.search(query.as_ref(), &fast_field_collector)?;
+            for part in &fast_field_values {
+                super::serialize::<i64, i64>(
+                    part.fast_field_values.as_slice(),
+                    &mut buffer,
+                    output_format,
+                    Some(part.partition_value),
+                )
+                .map_err(|_| {
+                    SearchError::InternalError(
+                        "Error when serializing i64 during export".to_owned(),
+                    )
+                })?;
+            }
+        }
+        (Type::U64, Type::U64) => {
+            let fast_field_collector = fast_field_collector_builder.build_u64_u64();
+            let fast_field_values = searcher.search(query.as_ref(), &fast_field_collector)?;
```

Comment:
all the blocking work like search & serialization must be spawn in a task with tokio `spawn_blocking` as it will hurt performance, see [docs](https://docs.rs/tokio/0.2.22/tokio/task/fn.spawn_blocking.html).